### PR TITLE
Chasm Fishing

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

image

Adds the capability of fishing in chasms.
Chasms contain Chasm Chrabs, which are cute little crustaceans you can eat or put in an aquarium.
More importantly, chasms contain everything which has ever fallen into a chasm.
Falling into a chasm is no longer instant permanent round removal... as long as someone is willing to do some fishing, which means your chances of return haven't necessarily increased that much.

image

Potential rescuers (or people lamenting about how they dropped their necropolis chest into a hole) should also beware, chasm crabs are actually young Lobstrosities and there's a rare but not insubstantial chance that you'll fish one of them up instead, which won't make it very happy.

If you somehow come back to life inside a chasm (generally only likely by being a changeling or some admin interference, but perhaps you have some kind of implant which does it) you'll climb back out again, so holes are no longer a way of disposing of a changeling for good. You think The Thing would be defeated if you threw it down a hole?
Ethereals can't manage this because falling into a chasm causes too much damage and they simply smash on the way down.

My current implementation for this involves chasms instantiating an abstract and invisible object when required to put things inside which as far as I can tell seems to work, but please do tell me this is a terrible idea if there's some side effects to this which I have not foreseen and I can try thinking of something else.

## Why It's Good For The Game

The primary reason for this PR is mostly that I think that it's funnier than how chasms currently work.
I also wanted more fishing content, and adding some weird ancilliary purpose for fishing rods seemed fun.
Instant deletion of a player's brain when they fell into a hole can be pretty deflating sometimes especially if they were a blood brother or something, now there's a way around that if you work for it.

## Changelog

:cl:
add: You can now fish in chasms, and might manage to catch things that careless miners have dropped in there.
add: If your pole is long enough to reach into a deep hole, you might catch crabs.
balance: Changelings can climb out of chasms if you throw them into one.
imageadd: An icon of a nice little crab, who will grow up into something less nice.
add: You can craft fishing equipment from various lavaland materials, so that Ashwalkers can fish up dead miners before you.
add: You can sometimes find worms while digging up tiles, if appropriate for that material.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
~~how does the fishing work? does it need to be on the same exact tile that they fell in? or does it grab from everything that has fallen into any chasm over the entire round?~~ if im reading the code correctly, it takes from all chasms, okay

if a changeling is thrown into the middle of a 3x3, how will they climb out? seeing as they'd be surrounded by chasms

**Comment by Jacquerel** [comment]
> ~how does the fishing work? does it need to be on the same exact tile that they fell in? or does it grab from everything that has fallen into any chasm over the entire round?~ if im reading the code correctly, it takes from all chasms, okay
> 
> if a changeling is thrown into the middle of a 3x3, how will they climb out? seeing as they'd be surrounded by chasms

When you climb out it then throws you in a random direction as you make a final leap for freedom (quite hard, you might break a limb).
It's far from impossible that you will simply land inside a _different_ hole and fall in, but that's your fault for getting thrown into a hole. You'll get out eventually unless the miners wall you in.

**Comment by timothymtorres** [comment]
Can you make some of the ghost roles that spawn on lavaland have access to a fishing pole?  It would fit this theme nicely.

**Comment by Shadowflame909** [comment]
This is indeed very cool. Perma-death stopped being cool 5 years ago!

**Comment by Wallemations** [comment]
How does this deal with having a GPS on you when you die? Suit sensors too

**Comment by GoldenAlpharex** [comment]
Now, imagine that it's a _grappling hook_ instead of a fishing rod. How even cooler that would be.

Cool idea to get rid of a very poor RR portion of the game, very flavorful.

**Comment by Jacquerel** [comment]
> 

If you can't already make a rod, line, and hooks from lavaland materials I will maybe make it so.

> How does this deal with having a GPS on you when you die? Suit sensors too

The positional information provided by the gps will point at the chasm you fell into.

**Comment by rasepretep** [comment]
A neat idea, but I have doubts about people going from chasm to chasm, fishing in them hoping to find the one miner that fell in one. Dying on Lavaland regardless of reason is already borderline RR since many miners don't bother carrying a GPS and even if they do, almost nobody checks on them until mats run out an hour later and the miner is already DNR.

Plus we already have a better way to make chasms not a threat in any way, and that's the Wormhole Jaunter worn in the belt slot.

**Comment by Jacquerel** [comment]
> A neat idea, but I have doubts about people going from chasm to chasm, fishing in them hoping to find the one miner that fell in one. Dying on Lavaland regardless of reason is already borderline RR since many miners don't bother carrying a GPS and even if they do, almost nobody checks on them until mats run out an hour later and the miner is already DNR.

With this system the pool of fishing results is the combined pool of anything which has fallen into any chasm, so you don't need to hunt the specific one.  
You ate right that this is merely upgrading "no chance of rescue" to "slim chance of rescue", but my view is if a miner brought neither a gps nor kheiral cuffs that is their own fault.

**Comment by rasepretep** [comment]
It's still a neat PR and I like it.

> but my view is if a miner brought neither a gps nor kheiral cuffs that is their own fault.

Also Wormhole Jaunter. I still can't believe almost nobody carries one.

**Comment by RazorManager89** [comment]
> add: If your pole is long enough to reach into a deep hole, you might catch crabs.

Give this man a medal

**Comment by Shadowflame909** [comment]
> It's still a neat PR and I like it.
> 
> > but my view is if a miner brought neither a gps nor kheiral cuffs that is their own fault.
> 
> Also Wormhole Jaunter. I still can't believe almost nobody carries one.

thats because it takes up a belt slot and doesnt work in a chest rig/webbing/backpack. The extra space is already used! No place for it.

File: code/datums/components/chasm.dm
Comment (SuperNovaa41) [inline code/datums/components/chasm.dm:43]: this should be done on `UnregisterFromParent`

File: code/modules/fishing/fish/fish_types.dm
Comment (SuperNovaa41) [inline code/modules/fishing/fish/fish_types.dm:185]:
```suggestion
/obj/item/fish/chasm_crab
```
a better path is easier to actually know what this is without going to the defintion

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

**Comment by Jacquerel** [comment]
> Can you make some of the ghost roles that spawn on lavaland have access to a fishing pole? It would fit this theme nicely.

I have added this but am considering whether I am now overloading this PR, if a maintainer desires I'll atomise it into a different one (although it may need to wait a bit as after this one I won't have any GBP left)

**Comment by rasepretep** [comment]
A quick question just popped in my head. Changelings or otherwise coming back to life inside the chasm, you climb back out. What if you fall in while wearing a Memento Mori?

**Comment by Jacquerel** [comment]
> A quick question just popped in my head. Changelings or otherwise coming back to life inside the chasm, you climb back out. What if you fall in while wearing a Memento Mori?

You die I'm afraid.
The Memento Mori only prevents you from entering crit, the chasm just kills you.

**Comment by StoneMason** [comment]
lavaland hermit should get a fishing rod and a little chasm that's blocked by railings

**Comment by Jacquerel** [comment]
> lavaland hermit should get a fishing rod and a little chasm that's blocked by railings

I like this idea but I don't want to add it in this PR.  
The changes already included should allow a hermit to make fishing supplies from lavaland materials.

**Comment by j-nengel** [comment]
chasms should just be removed

**Pull request merged**
